internal/operators/odf/common.go

```diff
@@ -0,0 +1,58 @@
+package odf
+
+import (
+	"errors"
+
+	"github.com/go-openapi/swag"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+)
+
+func getODFDeploymentMode(cluster *models.Cluster, odfMinimumHosts int64) odfDeploymentMode {
+	masterHosts, workerHosts, autoAssignHosts := common.GetHostsByEachRole(cluster, true)
+
+	masterCount := len(masterHosts)
+	workerCount := len(workerHosts)
+	autoAssignCount := len(autoAssignHosts)
+	hostCount := masterCount + workerCount + autoAssignCount
+
+	// To keep compatability with the behaviour until now.
+	if hostCount == common.MinMasterHostsNeededForInstallationInHaMode && workerCount == 0 {
+		return compactMode
+	}
+
+	if masterCount == hostCount && masterCount >= int(odfMinimumHosts) {
+		return compactMode
+	}
+
+	if masterCount >= common.MinMasterHostsNeededForInstallationInHaMode && workerCount >= int(odfMinimumHosts) {
+		return standardMode
+	}
+
+	// To be determined, the cluster is not yet in a valid form.
+	return unknown
+}
+
+// To keep compatability with the behaviour until now.
+// TODO - remove this once two control plane nodes OpenShift is implemented
+func isAutoAssignmentAllowed(cluster *models.Cluster) bool {
+	return len(cluster.Hosts) == common.MinMasterHostsNeededForInstallationInHaMode
+}
+
+// shouldHostRunODF returns whether a host will eventually run ODF worloads given the cluster it belongs to,
```

Comment:
I think we should specify that the function will return:
* nil if the ability of running ODF cannot be evaluated
* &true if the host  should run ODF
* &false otherwise

* error for invalid hosts